provisioning/provisioning-service-client/src/main/java/com/microsoft/azure/sdk/iot/provisioning/service/configs/DeviceRegistrationState.java

```diff
@@ -41,59 +42,68 @@
     private static final String REGISTRATION_ID_TAG = "registrationId";
     @Expose
     @SerializedName(REGISTRATION_ID_TAG)
+    @Getter
     private String registrationId;
 
     // the device identifier
     private static final String DEVICE_ID_TAG = "deviceId";
     @Expose
     @SerializedName(DEVICE_ID_TAG)
+    @Getter
     private String deviceId;
 
     // the created date and time
     private static final String CREATED_DATETIME_UTC_TAG = "createdDateTimeUtc";
     @Expose
     @SerializedName(CREATED_DATETIME_UTC_TAG)
     @SuppressWarnings("unused") // used by reflection during json serialization/deserialization
-    private String createdDateTimeUtc;
-    private transient Date createdDateTimeUtcDate;
+    private String createdDateTimeUtcString;
+    @Getter
+    private transient Date createdDateTimeUtc;
 
     // last update date and time
     private static final String LAST_UPDATED_DATETIME_UTC_TAG = "lastUpdatedDateTimeUtc";
     @Expose
     @SerializedName(LAST_UPDATED_DATETIME_UTC_TAG)
     @SuppressWarnings("unused") // used by reflection during json serialization/deserialization
-    private String lastUpdatedDateTimeUtc;
-    private transient Date lastUpdatedDateTimeUtcDate;
```

Comment:
Swapping the names of these fields a bit to maintain that deviceRegistrationState.getLastUpdatedDateTimeUtc() still returns a date rather than a string

Reply:
You'll see this same pattern in a few date time fields across a few serialization types

Reply:
this sure was confusing to get the flow -> getLastUpdatedDateTimeUtc()  returns Date lastUpdatedDateTimeUtcDate which it constructed out of a string lastUpdatedDateTimeUtc

Reply:
Yeah, the real unfortunate part of Lombok is that the getter's name is just get<FieldName> with no room for customization